clients/src/test/java/org/apache/kafka/common/requests/ProduceRequestTest.java

```diff
@@ -102,13 +103,36 @@ public void testBuildWithCurrentMessageFormat() {
         ProduceRequest.Builder requestBuilder = ProduceRequest.builder(
             new ProduceRequestData()
                 .setTopicData(new ProduceRequestData.TopicProduceDataCollection(Collections.singletonList(
-                    new ProduceRequestData.TopicProduceData().setName("test").setPartitionData(Collections.singletonList(
-                        new ProduceRequestData.PartitionProduceData().setIndex(9).setRecords(builder.build()))))
+                    new ProduceRequestData.TopicProduceData()
+                            .setTopicId(Uuid.fromString("H3Emm3vW7AKKO4NTRPaCWt"))
+                            .setPartitionData(Collections.singletonList(
+                                    new ProduceRequestData.PartitionProduceData().setIndex(9).setRecords(builder.build()))))
                     .iterator()))
                 .setAcks((short) 1)
                 .setTimeoutMs(5000),
             false);
-        assertEquals(3, requestBuilder.oldestAllowedVersion());
+        assertEquals(ApiKeys.PRODUCE.oldestVersion(), requestBuilder.oldestAllowedVersion());
+        assertEquals(ApiKeys.PRODUCE.latestVersion(), requestBuilder.latestAllowedVersion());
+    }
+
+    @Test
+    public void testBuildWithCurrentMessageFormatWithoutTopicId() {
+        ByteBuffer buffer = ByteBuffer.allocate(256);
+        MemoryRecordsBuilder builder = MemoryRecords.builder(buffer, RecordBatch.CURRENT_MAGIC_VALUE,
+                Compression.NONE, TimestampType.CREATE_TIME, 0L);
+        builder.append(10L, null, "a".getBytes());
+        ProduceRequest.Builder requestBuilder = ProduceRequest.builder(
+                new ProduceRequestData()
+                        .setTopicData(new ProduceRequestData.TopicProduceDataCollection(Collections.singletonList(
+                                        new ProduceRequestData.TopicProduceData()
+                                                .setName("topic")
```

Comment:
Could we add the following comment back?
// TopicId will default to Uuid.ZERO and client will get UNKNOWN_TOPIC_ID error.